apm-agent-core/src/main/java/co/elastic/apm/agent/impl/ElasticApmTracer.java

```diff
@@ -357,7 +358,7 @@ public void captureException(long epochMicros, @Nullable Throwable e, TraceConte
     }
 
     public void captureException(long epochMicros, @Nullable Throwable e, @Nullable TraceContextHolder<?> parent, @Nullable ClassLoader initiatingClassLoader) {
-        if (e != null) {
+        if (e != null && !WildcardMatcher.isAnyMatch(coreConfiguration.getIgnoreExceptions(), e.getClass().getName())) {
```

Comment:
Because this may be called frequently in some scenarios, what do you say we cache the result for each Exception class name and save some CPU? Then, if we want to apply this logic for inheritance as well, performance is not a concern anymore.

Reply:
Let's add the cache once we want to match on the hierarchy. For now, I don't think the cache is worth it.

Reply:
I've added a comment not to forget this when (and if) we need to implement this.